core/src/main/java/org/apache/accumulo/core/spi/common/ContextClassLoaderFactory.java

```diff
@@ -65,4 +65,15 @@ default void init(ContextClassLoaderEnvironment env) {}
    * @return the class loader for the given contextName
    */
   ClassLoader getClassLoader(String contextName);
+
+  /**
+   * Validate that the contextName is supported by the ContextClassLoaderFactory implementation
+   *
+   * @param contextName the name of the context that represents a class loader that is managed by
+   *        this factory (can be null)
+   * @return true if valid, false otherwise
+   */
+  default boolean isValid(String contextName) {
+    return false;
```

Comment:
Should the default be true to maintain existing behavior, or is existing behavior so harmful as to warrant making all contexts invalid by default?

Reply:
Based on my analysis in #3677 , an invalid context name could prevent any classes from being loaded, which could lead to the minor compaction thread for a tablet to die. From what I found, a minor compaction for a tablet will not occur again if the minor compaction thread dies, meaning that it can't be closed normally. I think the safest action when this occurs is a total restart of Accumulo.

The changes in #3677 try to prevent the minor compaction thread from dying, but there could be other areas within Accumulo where something similar happens when an invalid context name is used. Given that, and that the recovery for this situation is a total restart, I would suggest that `true` not be the default answer. In fact, I don't think this should be a default method, I think we should force all implementations to provide an answer.

Reply:
I see your point about making it stricter for safety... but making this default to `false` or making it a mandatory method would break things in a patch release on an LTM branch... and I don't think the benefits you describe outweigh the downsides of that breakage.

Reply:
I'm fine with bumping the minor version as part of this patch.

Reply:
That would be a non-LTM release, not part of the LTM release we are supporting as stable, though.

Reply:
So, we can't change the "current" LTM release?

Reply:
No, that wouldn't make sense to do. The 2.1 version was marked LTM, specifically because 2.1 is the version we wanted to communicate that we would invest our time patching to keep stable. That's what it means to be an LTM version. LTM is an attribute of a specific major.minor release series. If LTM follows us and moves across major/minor releases, then the whole LTM concept makes no sense and stops being useful to communicate that 2.1 is the version we're invested in.

Reply:
This is moot if we follow @keith-turner 's suggestion to use `getClassLoader(contextName)` as the default implementation, which I think is a good idea. If this returns something not null without throwing an exception, then this method should return true. If an implementation needs to avoid constructing the classloaders for some reason, then it can override the default method.